src/utils/permalinks/ElementPermalinkConstructor.ts

```diff
@@ -31,23 +31,32 @@ export default class ElementPermalinkConstructor extends PermalinkConstructor {
         }
     }
 
-    public forEvent(roomId: string, eventId: string, serverCandidates: string[]): string {
+    public forEvent(roomId: string, eventId: string, serverCandidates: string[], ispill = false): string {
+        if (ispill) {
+            return `https://matrix.to/#/${roomId}/${eventId}${this.encodeServerCandidates(serverCandidates)}`;
```

Comment:
This syntax is already generated by MatrixToPermalinkConstructor. I think probably the right fix here would be to pass the flag into `makeGenericPermalink` but then have `getPermalinkConstructor()` return the matrix.to one instead. This should make this change a lot simpler as all the various methods of the different PermalinkConstructor classes won't need the extra param.